src/main/java/org/cryptomator/fusecloudaccess/OpenFileFactory.java

```diff
@@ -47,10 +54,22 @@ public long open(Path path, Set<OpenFlags> flags) {
      * @param fileHandle file handle used to identify
      */
     public void close(long fileHandle) {
-        OpenFile file = openFiles.remove(fileHandle);
-        if (file != null) {
-            LOG.trace("Releasing {} {}", fileHandle, file);
+    	if( !isClosed.get()){
+            OpenFile file = openFiles.remove(fileHandle);
+            if (file != null) {
+                LOG.trace("Releasing {} {}", fileHandle, file);
+            }else {
+                LOG.trace("No open file with handle {} found.", fileHandle);
+            }
         }
     }
 
+    /**
+     * Closes this factory and all still open files
+     */
+    @Override
+    public void close() {
+        isClosed.set(true);
+        openFiles.clear();
+    }
```

Comment:
Why implement Closeable?

OpenFiles aren't Closeable and we don't occupy any resources. This just adds unnecessary complexity (such as the ClosedOpenFileFactoryException).

I'd revert this

Reply:
Resolved by 4cf3e917e28d5e269cb31bc124049f60b29422c1 and 11a649eabc1af168f2be325b6d1f6b18e72c74ce